OpenSim/Simulation/Test/testStatesTrajectory.cpp

```diff
@@ -717,18 +717,23 @@ void tableAndTrajectoryMatch(const Model& model,
 
     const auto& colNames = table.getColumnLabels();
 
+    SimTK::Vector stateValues; // working memory
+
     // Test that the data table has exactly the same numbers.
     for (size_t itime = 0; itime < states.getSize(); ++itime) {
         // Test time.
         SimTK_TEST(table.getIndependentColumn()[itime] ==
                    states[itime].getTime());
 
+        stateValues = model.getStateVariableValues(states[itime]);
+
         // Test state values.
         for (size_t icol = 0; icol < table.getNumColumns(); ++icol) {
             const auto& stateName = colNames[icol];
 
-            const auto& valueInStates = model.getStateVariableValue(
-                    states[itime], stateName);
+            const auto stateIndex = stateNames.findIndex(stateName);
```

Comment:
If the indices aren't going to change from State to State, you could create hold on to indices either as a flat list or a map keyed by name. `Array::findIndex()` is linear in number of elements and is doing string matching.